gno/p/role_manager/role_manager.gno

```diff
@@ -0,0 +1,406 @@
+package role_manager
+
+// TODO: RENAME FUNCTIONS (HAS PERMISSIONS -> UserHasPermission, HAS ROLE -> UserHasRole, etc)
+// TODO: CREATE HAS PERMISSION and HAS ROLE functions that just look if role or permission exists not if user has it
+import (
+	"std"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/ownable"
+)
+
+// well known roles
+const (
+	ROLE_USER = "user"
+)
+
+// well known permissions
+const (
+	PERM_ALL         = "all"
+	PERM_ADD_ROLE    = "add:role"
+	PERM_REMOVE_ROLE = "remove:role"
```

Comment:
doesn't this make more sense like `role:add`? so that they also display better when ordering globally.

Reply:
Yes indeed, i think have something like resources:actions could be nice like
``role:read,write,all`` with a way to create resources & actions. i will add this in V2, thanks for review :pray: